src/perfrecord.cpp

```diff
@@ -71,10 +75,51 @@ static QStringList sudoOptions(const QString& sudoBinary)
     return options;
 }
 
+static bool privsAlreadyElevated()
+{
+    struct ValueMap {
+        const char *path = nullptr;
+        int value;
+    };
+    static const std::array<ValueMap, 2> map = {{
+        {"/proc/sys/kernel/kptr_restrict", 0},
+        {"/proc/sys/kernel/perf_event_paranoid", -1},
+    }};
+
+    std::ifstream ifs;
```

Comment:
just move this into the lambda, then you can remove the `ifs.is_open` - or why do you want to reuse it? I don't see any reason for that

Reply:
So as not to construct two objects of the same type when I can re-use one object.